internal/pkg/client/cache/library.go

```diff
@@ -29,12 +32,25 @@ func LibraryImage(sum, name string) string {
 
 // LibraryImageExists returns whether the image with the SHA sum exists in the LibraryImage cache
 func LibraryImageExists(sum, name string) (bool, error) {
-	_, err := os.Stat(LibraryImage(sum, name))
+	imagePath := LibraryImage(sum, name)
+	_, err := os.Stat(imagePath)
 	if os.IsNotExist(err) {
 		return false, nil
 	} else if err != nil {
 		return false, err
 	}
 
+	cacheSum, err := client.ImageHash(imagePath)
+	if err != nil {
+		sylog.Debugf("Error getting ImageHash - Deleting Cached File: %v", err)
+		os.Remove(imagePath)
+		return false, nil
+	}
+	if cacheSum != sum {
+		sylog.Debugf("Cached File Sum(%s) and Expected Sum(%s) does not match - Deleting Cached File", cacheSum, sum)
+		os.Remove(imagePath)
```

Comment:
I don't think we should be removing files from the cache in this function, just checking if something exists or not with the correct checksum. I think `cmd/internal/cli/pull.go` is a good place to remove/overwrite an invalid cache entry and it could be done simply by changing the force option of `client.DownloadImage()` to `true`